packages/core/src/State.ts

```diff
@@ -21,6 +21,7 @@ import { StateNode } from './StateNode';
 import { getMeta, nextEvents } from './stateUtils';
 import { initEvent } from './actions';
 import { IS_PRODUCTION } from './environment';
+import { ExtraGenerics } from '.';
```

Comment:
Will `ExtraGenerics` work fine (not conflict) with the typegen work?

Reply:
Yeah, we'd just need to have another generic for the Typegen. If we start recommending `createModel` as the default it means this isn't so user-facing.

Using `ExtraGenerics` as an object means we only need to do this refactor once, too - we can include any other optional generics on the `ExtraGenerics` object